src/Settings.cpp

```diff
@@ -26,8 +28,15 @@ QVariant Settings::getValue(const std::string& group, const std::string& name)
         return cacheIndex->second;
     } else {
         // Nothing found in the cache, so get the value from the settings file or get the default value if there is no value set yet
-        QSettings settings(QApplication::organizationName(), QApplication::organizationName());
-        QVariant value = settings.value(QString::fromStdString(group + "/" + name), getDefaultValue(group, name));
+        QSettings* settings = NULL;
+
+        if(userConfiguration)
+            settings = new QSettings(userConfigurationDir, QSettings::IniFormat);
+        else
+            settings = new QSettings(QCoreApplication::organizationName(), QCoreApplication::organizationName());
+
+        QVariant value = (*settings).value(QString::fromStdString(group + "/" + name), getDefaultValue(group, name));
+        delete settings;
```

Comment:
Now that we use a pointer, it would be a good idea to construct the settings object only once. I think saving preferences would be much more efficient in that case, but I don't know why it worked like that before, maybe I'm missing something.